libraries/standard/mqtt/include/mqtt.h

```diff
@@ -35,14 +39,9 @@
  */
 #define MQTT_PACKET_ID_INVALID    ( ( uint16_t ) 0U )
 
-struct MQTTPubAckInfo;
-typedef struct MQTTPubAckInfo         MQTTPubAckInfo_t;
-
+/* Forward declared structs for the MQTTEventCallback_t. */
```

Comment:
If we're able to forward declare the structs but not the typedefs, why not declare all of the structs here, instead of just the ones in the callback? It would let us easily see at the top of the file what structs are used

Reply:
Because the context struct needs the definition of the MQTTEventCallback_t

Reply:
Adding the third struct per misread confusion.